torch/_export/non_strict_utils.py

```diff
@@ -339,21 +340,25 @@ def make_constraints(
             continue
         shape_spec = flat_dynamic_shapes[input_index - num_lifted_inputs]
         for i, d in enumerate(node.meta["val"].shape):
-            if isinstance(d, torch.SymInt):
+            if isinstance(d, torch.SymInt) and not d.node.expr.is_number:
```

Comment:
What happens when `d.node.expr.is_number`? Maybe a positive test would be better here, are you expecting it to be a symbol or are you worried it could be an expression containing symbols?

Reply:
Also looking at my comment in the code, I'm wondering if `_expr` could be a symbol but `expr` is a number, maybe because a replacement exists? Not sure, babbling here, maybe you observed something that can help refresh my memory?

Reply:
It seems like auto dynamic mode, when things specialize, will create a torch.symint instead of a concrete int